sihl/src/database.ml

```diff
@@ -299,12 +332,15 @@ let used_database () =
 
 (* Service lifecycle *)
 
-let start () =
-  (* Make sure that configuration is valid *)
-  Core_configuration.require schema;
-  (* Make sure that database is online when starting service. *)
-  let _ = fetch_pool () in
-  Lwt.return ()
+let start ?(skip_default_pool_creation = false) () =
```

Comment:
I just remembered: A sihl user does not call this function. This param should probably come from `Database.register` no?

Reply:
You are right, let's just read an env var `DATABASE_SKIP_DEFAULT_POOL`

Reply:
Is an env var the best approach here? Maybe passing it in as an arg is better? I'm afraid of overloading Sihl with env vars...

Reply:
With `sihl config:list` we have a solid system in place to list all options and this is the main way to configure services. We only pass data to services using register, not configuration.